source/extensions/filters/http/lua/lua_filter.cc

```diff
@@ -264,20 +254,58 @@ int StreamHandleWrapper::luaRespond(lua_State* state) {
 int StreamHandleWrapper::luaHttpCall(lua_State* state) {
   ASSERT(state_ == State::Running);
 
+  luaL_checktype(state, 3, LUA_TTABLE);
+  size_t body_size;
+  const char* body = luaL_optlstring(state, 4, nullptr, &body_size);
```

Comment:
I'm not familiar with the LUA API. Are these 3, 4, 5 some sort of indexes? If so can you create named constants for the numbers, please?

Reply:
Those are indexes, yes. Index of value in the stack. The current codebase has it all using numbers.